### ISSUE TITLE
Reduce number of required options

### ISSUE THREAD

**Issue by Theo47** [Lyle-Tafoya/Steam-Categorizer#8]
Some recommendations to minimize the number of options the user needs to provide.
1. **--key:** You could consider parsing the owned game list from localconfig.vdf instead to remove the need for providing an api key.
2. **--id:** Apparently if you get the bolded part of the path to sharedconfig.vdf (/home/my_username/.steam/steam/userdata/**65357252**/7/remote/sharedconfig.vdf) and add 0x0110000100000000 to it, you end up with the ID64. You could use that instead of asking the user to provide an id too.
3. **--config:** If the user provided an id then you can do the opposite here. I am not sure what the process is for converting Steam Id to Steam ID64 but after you do that you can substract 0x0110000100000000 from it to get the number you need for the path. For the path to steam you can try checking if it is located to the default install path. For windows it is "C:\Program Files\Steam" or "C:\Program Files (x86)\Steam" depending or whether the system is 32-bit or 64-bit. I don't know what the default install path for linux and macos is but a google search would probably provide you with that. You could also add a --path-to-steam option for the cases where steam isn't installed on the default path.
4. **--output:** You could probably set a default for this to be the same as --config if you haven't already done so. You could also have it create a backup before overwriting the file if you are worried for errors.
5. **--preferences:** You could add a default path for this too if you haven't done so already. For windows it would probably be the location the program was run from. I can't speak for linux.
6. **--birthdate:** I assume in this case we only care about one thing. Whether we pass the age check or not. The actual birthdate is not that important and i assume most people don't know their birthdate in epoch date and would just use "-b 0" anyway. So maybe just make one case the default and add an option for the other. As in make not bypassing age check the default and add a --bypass-age-check option or the opposite. Personally i'd prefer the latter.

**Comment by Lyle-Tafoya** [comment]
These are some great suggestions. I will do my best to implement them. Thank you for taking the time to provide your input. It is very valuable to me, especially during these early stages of development.

**Comment by Lyle-Tafoya** [comment]
Just to clarify some things:
- The localconfig.vdf does not seem to contain a complete list of appids. Every section of that config I tried to parse out either had either more or less keys than I have games in my library. I'm looking into the possibility of parsing this information out from https://steamcommunity.com/id/user_name/games, However I am currently receiving a 500 internal server error while trying to view that page in my web browser, so I am blocked on that for now.
- Regarding default install paths for Steam, I am trying to avoid having to write any operating system specific code as I do not have a Windows computer to test Windows-specific changes in. For this reason, I might always require that the user point this tool to the right directory if nothing else.
- The Steam-Categorizer preferences file will eventually be generated on the fly by the time we reach the end-user release. First it will fetch all the tags from the store pages, then it will compile a list and give the user an opportunity to map out the tags to category names using a point and click interface before it applies them. This will remove the need to explicitly pass in a config file, since it will be created from scratch by Steam-Categorizer during this step.
- Auto-backup functionality is planned. I may even use a local Git repository to manage different versions of the sharedconfig.vdf. This will allow for easy rollback to any previous version if needed. I'll put in a separate issue/ticket for this.

**Comment by Theo47** [comment]
- I got the localconfig.vdf idea (among most of the others) from how depressurizer handles things so it should be possible. Perhaps one of the lists also includes dlcs and depressurizer just filters them out. I know it would be easy for it since it already stores that information but it may be a problem for you. Or i may have given you the wrong filename. I'll try to check the code later to give you a better answer about how that works.
- I see. Then perhaps once you reach the gui phase you could also add a settings file so that the user would only need to point you to the steam path once.
- You may still want to save the preferences file and load it in next run though instead of having to recreate it from scratch. It doesn't seem to be fast to create and chances are if you renamed "2D Fighter" to [ "Fighting" ] once then you are going to do so again. So first load it and then give the user a chance to edit it (using the GUI)?

**Comment by Theo47** [comment]
Took a look at the code and turns out it is more complex than what i thought when i suggested it.

It reads from the "Licenses" section of localconfig.vdf. But that contains package ids not game ids.
A package may contain more than 1 games as well as dlcs. You can take a look at [340](http://store.steampowered.com/sub/340/) and [49800](http://store.steampowered.com/sub/49800/) to better see what that means.

It then reads from steam/appcache/packageinfo.vdf using a binary reader to get the packageid to gameid conversions.

Finally it filters out all dlcs. I think it does so by reading from a third local file to get the app type of each gameid but i didn't actually check.

I guess reading from [https://steamcommunity.com/id/user_name/games](https://steamcommunity.com/id/user_name/games) would be easier to implement.

**Comment by Lyle-Tafoya** [comment]
I may consider following that same process, in order to keep as much work done offline as possible. However, that may be more of a long term goal since I have already determined how to parse the steamcommunity games page and have added logic to my development branch to skip over "Downloadable Content" titles whenever I am parsing the store page for them.

If there is a way to determine whether a game is downloadable content offline, I may look into that. I have plans to do a little research on all the Steam .vdf files to determine what their potential usefulness may be to this project.

**Comment by Lyle-Tafoya** [comment]
This issue is solved as of v0.5.0 97d799fc1298671cf43530dd9dec8506c62b31e4
The sharedconfig.vdf file and steam url name can be loaded from a preferences file
The following options have been removed completely: --key, --id, --output, --birthdate
